### PR TITLE
Documentation Update

### PR DESCRIPTION
**Pull request by tanmoysrt**
#### Describe the changes you have made in this PR -
Revamp Documentation
- For Docker Development Environment [Windows, Linux, Mac]
- For Manual Setup [Windows, Linux, Mac]
- For Remote Development [Gitpod, Github Codespaces]

Note: Please check **Allow edits from maintainers.** if you would like us to assist in the PR.

### CODE REVIEW

File: SETUP.md [hunk 1]
```diff
@@ -5,112 +5,88 @@ Please go through the [Contribution Guidelines](CONTRIBUTING.md) before going fo
...
+---
+
 ## Installation
 There are several ways to run your own instance of CircuitVerse:
 
-### Gitpod Cloud Environment
-[Gitpod](https://www.gitpod.io/) is a free platform that allows you to develop CircuitVerse in a cloud VS Code environment.
+| Method | Operating System | Documentation |
+| --- | --- | --- |
+| GitHub Codespaces | Any | [Click Here](https://github.com/tanmoysrt/CircuitVerse/tree/master/installation_docs/remote_development.md#github-codespaces) |
```
Comment (tachyons) [inline SETUP.md:15]: Use links to Circuitverse repo, not the personal repo

File: DEBUGGER-SETUP.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
+## Instructions For Using Ruby Debugger
+
```
Comment (vedant-jain03) [inline DEBUGGER-SETUP.md:2]: Can we give some intro for ruby debugger

File: DEBUGGER-SETUP.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+#### Debugging with VSCode
+1. Install [VSCode rdbg Ruby Debugger](VSCode rdbg Ruby Debugge) extenstion.
```
Comment (vedant-jain03) [inline DEBUGGER-SETUP.md:4]: Link missing

File: DEBUGGER-SETUP.md [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+#### Debugging with Chrome
+
+1. Run the app by `./bin/dev chrome_debug`
+2. Chrome Devtools will be open
+3. In Chrome, Move to `Filesystem` and add `CircuitVerse` folder to workspace
+4. Now you can open any file and set breakpoint to debug
+
+#### Debugging with VSCode (Docker)
+1. Install [VSCode rdbg Ruby Debugger](VSCode rdbg Ruby Debugge) extenstion.
```
Comment (vedant-jain03) [inline DEBUGGER-SETUP.md:19]: missing link

File: SETUP.md [hunk 5]
```diff
@@ -5,112 +5,88 @@ Please go through the [Contribution Guidelines](CONTRIBUTING.md) before going fo
...
+# Enable features (:recaptcha, :forum)
+> Flipper.enable :recaptcha
 
-```sh
-git clone -c core.symlinks=true https://github.com/CircuitVerse/CircuitVerse.git
-cd CircuitVerse
+# Disable features (:project_comments, :lms_integration)
+> Flipper.disable :forum
 ```
+Flipper dashboard can be accessed at - http://localhost:3000/flipper/ from where following features can be enabled/disabled.
```
Comment (vedant-jain03) [inline SETUP.md:56]: Add a note or something that flipper route is for admins

File: SETUP.md [hunk 6]
```diff
@@ -5,112 +5,88 @@ Please go through the [Contribution Guidelines](CONTRIBUTING.md) before going fo
...
+
+## Deployment Guide
+### Heroku Deployment
+[Heroku](https://www.heroku.com) is a free cloud platform that can be used for deployment of CircuitVerse
```
Comment (vedant-jain03) [inline SETUP.md:60]: It is not free anymore

**Review by vedant-jain03 — CHANGES_REQUESTED** [review]

File: SETUP.md (hunk 6 above)
Comment (tachyons) [reply SETUP.md:68]: we can remove docker now

File: DEBUGGER-SETUP.md [hunk 7]
```diff
@@ -0,0 +1,25 @@
+## Instructions For Using Ruby Debugger
+Ruby's deb.rb is a powerful tool that allows for you to stop the execution of the application at a particular moment and to investigate and interact within that context
```
Comment (tachyons) [inline DEBUGGER-SETUP.md:2]: deb.rb ?

File: installation_docs/manual/linux.md [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+- Ruby 3.2.1
+- [Redis 7.0 [atleast]](https://redis.io/docs/getting-started/installation/install-redis-on-linux/)
+- [ImageMagick](https://imagemagick.org/) - Image manipulation library
+- [Node.js 16.x](https://nodejs.org/it/download)
+- [Yarn](https://yarnpkg.com/getting-started/install)
+- [CMAKE](https://cmake.org/install/)
+- OpenSSL
+- libpq-dev
+- [PostgreSQL](https://www.postgresql.org/) (`12`) - Database
+
```
Comment (tachyons) [inline installation_docs/manual/linux.md:29]: Add `apt` command to download all dependencies for ubuntu and debian, For ruby use `rvm` or `asdf` guide to install the correct version of ruby

File: installation_docs/manual/mac.md [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+- [Git](https://git-scm.com/) - using a GUI such as [SourceTree](https://www.sourcetreeapp.com/) or [GitHub Desktop](https://desktop.github.com/) can help
+- [RVM](https://rvm.io/rvm/install) 
+- Ruby 3.2.1
+- [Redis 7.0 [atleast]](https://redis.io/docs/getting-started/installation/install-redis-on-linux/)
+- [ImageMagick](https://imagemagick.org/) - Image manipulation library
+- [Node.js 16.x](https://nodejs.org/it/download)
+- [Yarn](https://yarnpkg.com/getting-started/install)
+- [CMAKE](https://cmake.org/install/)
+- OpenSSL
+- [PostgreSQL](https://www.postgresql.org/download/macosx/) (`12`) - Database
```
Comment (tachyons) [inline installation_docs/manual/mac.md:27]: We can add brewfile to simplify this https://thoughtbot.com/blog/brewfile-a-gemfile-but-for-homebrew

**Comment by tachyons** [comment]
Can we adapt this script ? https://github.com/tramlinehq/tramline/blob/main/bin/setup.mac

File: DEBUGGER-SETUP.md (hunk 7 above)
Comment (tanmoysrt) [reply DEBUGGER-SETUP.md]: typo mistake . Fixed !

File: installation_docs/manual/linux.md (hunk 8 above)
Comment (tanmoysrt) [reply installation_docs/manual/linux.md:73]: updated linux documentation with respective commands of each dependency

File: installation_docs/manual/mac.md (hunk 9 above)
Comment (tanmoysrt) [reply installation_docs/manual/mac.md]: yes added this

**Comment by tanmoysrt** [comment]
> Can we adapt this script ? https://github.com/tramlinehq/tramline/blob/main/bin/setup.mac

yes will check it. I  need to work with the mentor as I don't have access to mac system for verify setup

File: SETUP.md [hunk 10]
```diff
@@ -5,112 +5,87 @@ Please go through the [Contribution Guidelines](CONTRIBUTING.md) before going fo
...
+# Disable features (:project_comments, :lms_integration)
+> Flipper.disable :forum
 ```
-If you are cloning on Windows machine, use following command with an **administrative shell** to clone the repo.
+Flipper dashboard can be accessed at - http://localhost:3000/flipper/ from where following features can be enabled/disabled.
 
-```sh
-git clone -c core.symlinks=true https://github.com/CircuitVerse/CircuitVerse.git
-cd CircuitVerse
+**Note :** User need to login as admin first, then only Flipper dashboard can be accessed
```
Comment (tachyons) [inline SETUP.md:58]:
```suggestion
**Note :** User need to log in as admin first, then only Flipper dashboard can be accessed
```

**Comment by tachyons** [comment]
Could you check language check? Most of them can be ignored as they are spell cheking code

**Comment by tanmoysrt** [comment]
> Could you check language check? Most of them can be ignored as they are spell cheking code

Sure! Wil look at that

**Comment by tanmoysrt** [comment]
reviewdog is not giving any error. However giving some warning , that's  due to checking spelling in code sections.

**Pull request merged**